src/prefect/task_engine.py

```diff
@@ -631,12 +599,6 @@ def setup_run_context(self, client: Optional[SyncPrefectClient] = None):
                     task=self.task, parameters=self.parameters
                 )
 
-                if not PREFECT_EXPERIMENTAL_ENABLE_CLIENT_SIDE_TASK_ORCHESTRATION:
-                    # update the task run name if necessary
-                    self.client.set_task_run_name(
```

Comment:
when does the task run name get set now?

Reply:
The events pipeline updates the task run name after it's set on the engine object 👍